k2/csrc/ragged_utils.cu

```diff
@@ -0,0 +1,182 @@
+/**
+ * @brief
+ * ragged
+ *
+ * @copyright
+ * Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey, Haowen Qiu)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#include <cub/cub.cuh>
+#include <vector>
+
+#include "k2/csrc/array_ops.h"
+#include "k2/csrc/macros.h"
+#include "k2/csrc/math.h"
+#include "k2/csrc/ragged.h"
+#include "k2/csrc/ragged_utils.h"
+#include "k2/csrc/ragged_ops.h"
+
+namespace k2 {
+
+
+void CheckAxisEqual(int32_t num_srcs,
+                    int32_t axis,
+                    RaggedShape **src) {
+  if (num_srcs <= 1)
+    return;
+  K2_CHECK(axis >= 0 && axis + 1 < src[0]->NumAxes());
+  std::vector<int32_t*> row_splits_data_vec;
+  row_splits_data_vec.reserve(num_srcs);
+  int32_t row_splits_dim, row_ids_dim;
+  for (int32_t s = 0; s < num_srcs; s++) {
+    // RowSplits(1) .. is the lowest numbered row-splits...
+    int32_t *data = src[s]->RowSplits(axis + 1).Data();
+    if (s == 0 || data != row_splits_data_vec[0])
+      row_splits_data_vec.push_back(data);
+    if (s == 0) {
+      row_splits_dim = src[s]->RowSplits(axis + 1).Dim();
+      row_ids_dim = src[s]->RowIds(axis + 1).Dim();
+    } else {
+      K2_CHECK_EQ(row_splits_dim, src[s]->RowSplits(axis + 1).Dim());
+      K2_CHECK_EQ(row_ids_dim, src[s]->RowIds(axis + 1).Dim());
+    }
+  }
+  if (row_splits_data_vec.size() <= 1)
```

Comment:
maybe it's better to add documentation to say those srcs are exactly same (same address of row_splits) on that aixs before this line?